src/main/java/org/opensearch/knn/jni/FaissService.java

```diff
@@ -49,6 +49,20 @@ class FaissService {
         });
     }
 
+    /**
+     * Create a binary index for the native library The memory occupied by the vectorsAddress will be freed up during the
+     * function call. So Java layer doesn't need to free up the memory. This is not an ideal behavior because Java layer
+     * created the memory address and that should only free up the memory. We are tracking the proper fix for this on this
+     * <a href="https://github.com/opensearch-project/k-NN/issues/1600">issue</a>
+     *
+     * @param ids array of ids mapping to the data passed in
+     * @param vectorsAddress address of native memory where vectors are stored
+     * @param dim dimension of the vector to be indexed
+     * @param indexPath path to save index file to
+     * @param parameters parameters to build index
+     */
+    public static native void createBinaryIndex(int[] ids, long vectorsAddress, int dim, String indexPath, Map<String, Object> parameters);
```

Comment:
I don't think so we need this. Can we not use the parameters object to know if its a binary index or normal index.

Reply:
I believe such logic should be in Java side to make life easier. Also, the difference between binary index and non binary index is more than what it looks like at first glance. If we merge the two method, there will be lots of if/else and lots of different type of variables which will be used either by binary index or non binary index only.

Reply:
> If we merge the two method, there will be lots of if/else and lots of different type of variables

I would want to understand this further. Shouldn't we be able to avoid/minimize if-else by introducing an interface? 

```
- Convert everything to cpp type upfront
- Call the factory which can decide on the type of index and execute the operations required
- delete references and return results
```

Reply:
The parameter seems same but actually it is not because what we have in vectorsAddress is different between binary index and non binary index.

Also, the factory might end up having two different method(like what we have now) to create index at the end anyway.